subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/result/ComponentSelectorSerializer.java

```diff
@@ -84,12 +84,13 @@ ImmutableVersionConstraint readVersionConstraint(Decoder decoder) throws IOExcep
         String requires = decoder.readString();
         String prefers = decoder.readString();
         String strictly = decoder.readString();
+        String branch = decoder.readNullableString();
```

Comment:
If we have to do this, then we've triggered an unnecessary deserialise that wasn't happening before. We should instead get rid of the extra deserialise.

We should also add a separate issue to fix `ResolveableDependencies.getResolutionResult()` when there are source dependencies in the graph, but keep that separate.

Reply:
I reverted this change.